docs: add app directory structure conventions (#19)
moxystudio/react-native-with-moxy: docusaurus/docs/welcome/conventions.md

The folder-structure convention favours co-location of assets and their requesters. Here's how it looks like:

```
├── package-lock.json
├── package.json
└── src
    ├── app
    │   ├── App.js

>     │   ├── App.test.js

Reviewer: Shoudn't the last leaf be connected with └── instead of ├── ?

    ├── navigation
    │   ├── index.js
    │   ├── index.test.js
    │   ├── profile-stack
    │   │   ├── header
    │   │   ├── index.js
    │   │   └── index.test.js
    │   └── root
    │       ├── index.js
    │       └── index.test.js
    ├── screens
    │   ├── home
    │   │   ├── HomeScreen.js
    │   │   ├── HomeScreen.test.js
    │   │   ├── header
    │   │   └── styles

>     │   │       ├── index.js

Reviewer: Shoudn't the last leaf be connected with └── instead of ├── ?

    │   └── profile
    │       ├── ProfileScreen.js
    │       ├── ProfileScreen.test.js
    │       └── styles

>     │           ├── index.js

Reviewer: Shoudn't the last leaf be connected with └── instead of ├── ?

    └── shared
        ├── components
        ├── media
        │   ├── fonts
        │   └── icons
        ├── styles
        └── test-utils
            └── components
```

...where:

- `src`: Where the code for your application will be.
    - `app`: Where your App component will be.
    - `navigation`: Where you can define your navigators and corresponding routes for your application.

>         - `index.js`: This file is where the main stack is defined and it imports all other navigators

Reviewer: ```suggestion
        - `index.js`: This file is where the root stack is defined and other navigators are imported in.
```

>         - `profile-stack`: This is an example of a navigator that you'll reference in the main stack and the structure you'll find inside it. The name always contains the type of navigator we are creating (in this case, stack). Commonly, you'll find the following files / folders here:

Reviewer: ```suggestion
        - `profile-stack`: This is an example of a navigator that you'll reference in the root stack and the structure you'll find inside it. The name always contains the type of navigator we are creating (in this case, stack). Commonly, you'll find the following files / folders here:
```

            - `index.js`: This is your navigator file.
            - `index.test.js`: This is the test file corresponding to this navigator.
            - `header`: Sometimes a navigator may have an header different from the rest of the app. This folder is where you create that header and will contain a structure similar to a screen (`ProfileStackHeader.js`, `ProfileStackHeader.test.js`, `index.js`, `styles`). Default headers are set in the Navigator creation under options.
    - `screens`: Where you can store the source code of your screens. The screens are referenced in the navigator so you will be getting the screen files to the navigation route component from here.
        - `home`: This an example of a component and the structure you'll find inside. Commonly, you'll find the following files here:
            - `index.js`: This file will simply export your component file.
            - `Home.js`: This is your component file.
            - `Home.test.js`: This is the test file corresponding to this component.

>             - `header`: Similar to what happens in navigation this folder is where we define a custom header for this screen. The structure will be the same as in the navigation. Custom headers for a specific screen are set in the screen render through the navigation prop `navigation.setOptions({ header: () => <HomeHeader /> })`

Reviewer: ```suggestion
            - `header`: Similar to what happens in navigation, this folder is where we define a custom header for this screen. The structure is the same as in the navigation. Custom headers for a specific screen are set while rendering the screen through the navigation prop as follows: `navigation.setOptions({ header: () => <HomeHeader /> })`.
```

>             - `styles`: This is where you'll have styles that are only used in this component. As in React-Native we do not have css, the folder contains a sinngle `index.js` file which exports the styles.

Reviewer: ```suggestion
            - `styles`: This is where you'll have styles that are only used in this component. As in React Native there is no CSS, this folder contains a single `index.js` file which exports the styles.
```

            - `components`: If you have abstracted parts of this component into smaller components, use this folder to co-locate them.
    - `shared`: Where you can put content that is shared between pages and cannot be directly co-located with its interested parties.
        - `components`: The folder for components that are shared between pages. You can find an example just above of what files each component is expected to have.
        - `media`: The folder for media (images, fonts, etc.) that is shared between many components.
            - `fonts`: This is where you can store your font files.
            - `images`: This is where you can store image files.

>         - `styles`:  The folder for `.js` files that will be spread through context using `ThemeContext`. This contains mostly what we have in the web: `mixins`, `typography`, `colors` with a slight twist on its definitions as it is `js` only.

Reviewer: ```suggestion
        - `styles`:  The folder for `.js` files that is passed down through the components tree  with `ThemeContext`. This contains mostly what we have in the web: `mixins`, `typography`, `colors` with a slight twist on its definitions as it is JavaScript only.
```
